MapboxNavigationUI/NavigationMapView.swift

```diff
@@ -35,16 +32,16 @@ open class NavigationMapView: MGLMapView {
             style.removeSource(source)
         }
         
-        let polyline = shape(for: route)
+        let polyline = shape(describing: route)
         
         if let source = style.source(withIdentifier: sourceIdentifier) as? MGLShapeSource {
             source.shape = polyline
         } else {
-            let geoJSONSource = MGLShapeSource(identifier: sourceIdentifier, shape: polyline, options: nil)
-            style.addSource(geoJSONSource)
+            let lineSource = MGLShapeSource(identifier: sourceIdentifier, shape: polyline, options: nil)
+            style.addSource(lineSource)
             
-            let line = lineStyle(MGLLineStyleLayer(identifier: routeLayerIdentifier, source: geoJSONSource))
-            let lineCasing = lineCasingStyle(MGLLineStyleLayer(identifier: routeLayerCasingIdentifier, source: geoJSONSource))
+            let line = routeStyleLayer(identifier: routeLayerIdentifier, source: lineSource)
```

Comment:
Now that these method names are more descriptive, it might be worth adding a NavigationMapViewDelegate protocol with optional methods like `mapView(_:routeStyleLayerWithIdentifier:source:)`, and having this code call those methods. We would fall back to the `routeStyleLayer(identifier:source:)` method implemented here if those methods aren’t implemented.

This PR can land without the delegate protocol, but if subclassing turns out to be problematic, this is the solution.